x-pack/plugins/cross_cluster_replication/__jest__/client_integration/auto_follow_pattern_list.test.js

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import sinon from 'sinon';
+
+import { initTestBed, mockAllHttpRequests, nextTick, getRandomString, findTestSubject } from './test_helpers';
+import { AutoFollowPatternList } from '../../public/app/sections/home/auto_follow_pattern_list';
+import { getAutoFollowPatternClientMock } from '../../fixtures/auto_follow_pattern';
+
+jest.mock('ui/chrome', () => ({
+  addBasePath: () => 'api/cross_cluster_replication',
+  breadcrumbs: { set: () => {} },
+}));
+
+jest.mock('ui/index_patterns', () => {
+  const { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE } = require.requireActual('../../../../../src/legacy/ui/public/index_patterns/constants'); // eslint-disable-line max-len
+  return { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE };
+});
+
+describe('<AutoFollowPatternList />', () => {
+  let server;
+  let find;
+  let exists;
+  let component;
+  let getMetadataFromEuiTable;
+  let getUserActions;
+  let tableCellsValues;
+  let rows;
+  let updateHttpMockResponse;
+
+  beforeEach(() => {
+    server = sinon.fakeServer.create();
+    server.respondImmediately = true;
+    (updateHttpMockResponse = mockAllHttpRequests(server));
+  });
+
+  describe('on component mount', () => {
+    beforeEach(async () => {
+      ({ exists } = initTestBed(AutoFollowPatternList));
+    });
+
+    test('should show a loading indicator on component', async () => {
+      expect(exists('ccrAutoFollowPatternLoading')).toBe(true);
+    });
+  });
+
+  describe('when there are no auto-follow patterns', () => {
+    beforeEach(async () => {
+      ({ exists, component } = initTestBed(AutoFollowPatternList));
+
+      await nextTick(); // We need to wait next tick for the mock server response to comes in
+      component.update();
+    });
+
+    test('should display an empty prompt', async () => {
+      expect(exists('ccrAutoFollowPatternEmptyPrompt')).toBe(true);
+    });
+
+    test('should have a button to create a follower index', async () => {
+      expect(exists('ccrCreateAutoFollowPatternButton')).toBe(true);
+    });
+  });
+
+  describe('when there are auto-follow patterns', async () => {
+    // For deterministic tests, we need to make sure that autoFollowPattern1 comes before autoFollowPattern2
+    // in the table list that is rendered. As the table orders alphabetically by index name
+    // we prefix the random name to make sure that autoFollowPattern1 comes before autoFollowPattern2
+    const testPrefix = 'prefix_';
+    const testSuffix = '_suffix';
+
+    const autoFollowPattern1 = getAutoFollowPatternClientMock({
+      name: `a${getRandomString()}`,
```

Comment:
Short answer: no need to think about names! 😄 
Long answer: I fought many times bugs in my tests because I gave the same name in multiple places.. so now I try to never go that road again whenever possible.  Writing`'foo'` or `getRandomString()` is the same to me. With the later, I sleep better 😊